diff --git a/lib/time_realignment_ff_impl.cc b/lib/time_realignment_ff_impl.cc
index bca3a4f..72f52db 100644
--- a/lib/time_realignment_ff_impl.cc
+++ b/lib/time_realignment_ff_impl.cc
@@ -179,7 +179,7 @@ namespace gr {
             if( d_not_found_stamp_utc == 0)
                 d_not_found_stamp_utc = get_timestamp_nano_utc();
 
-            if( d_not_found_stamp_utc != 0 && abs( get_timestamp_nano_utc() - d_not_found_stamp_utc ) > d_max_buffer_time_ns )
+            if( d_not_found_stamp_utc != 0 && abs( static_cast<long long int>( get_timestamp_nano_utc() - d_not_found_stamp_utc ) ) > d_max_buffer_time_ns )
             {
                 d_not_found_stamp_utc = 0; //reset stamp
                 GR_LOG_ERROR(d_logger, name() + ": No WR-Tag found for trigger tag after waiting " + std::to_string(get_max_buffer_time())+ "s. Trigger will be forwarded without realligment. Possibly max_buffer_time needs to be adjusted." );
